packages/dds/tree/src/simple-tree/api/schemaFactory.ts

```diff
@@ -352,16 +361,22 @@ export class SchemaFactory<
 	 * class NamedMap extends factory.map("name", factory.number) {}
 	 * ```
 	 */
-	public map<Name extends TName, const T extends ImplicitAllowedTypes>(
+	public map<
+		Name extends TName,
+		const T extends ImplicitAllowedTypes,
+		const TCustomMetadata = unknown,
+	>(
 		name: Name,
 		allowedTypes: T,
+		props?: NodeSchemaProps<TCustomMetadata>,
```

Comment:
see comment I made on the structural array API: unless I'm missing something this seems impossible to define a good behavior for.